### PR TITLE
Manifest file changes

### PR DESCRIPTION
**Pull request by rombie**

### CODE REVIEW

File: cluster/contrail-api.manifest [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	"containers":[{
 	    "name": "contrail-api",
 	    "image": "opencontrail/config:2.20",
-	    "command": ["/bin/sh", "-c", "/usr/bin/contrail-api --conf_file=/etc/contrail/contrail-api.conf 1>>/var/log/contrail/contrail-api.stdout 2>>/var/log/contrail/contrail-api.err"],
+	    "command": ["/bin/sh", "-c", "/usr/bin/contrail-api --conf_file=/etc/contrail/contrail-api.conf --disc_server_ip=127.0.0.1 1>>/var/log/contrail/contrail-api.stdout 2>>/var/log/contrail/contrail-api.err"],
```
Comment (pedro-r-marques) [inline cluster/contrail-api.manifest]: There isn't an advantage to run discovery in this setups, correct ?

File: cluster/contrail-control.manifest [hunk 2]
```diff
@@ -7,7 +7,7 @@
 	"containers":[{
 	    "name": "contrail-control",
 	    "image": "opencontrail/control:2.20",
-	    "command": ["/bin/bash", "-c", "echo 127.0.0.1 $(hostname) >> /etc/hosts && /usr/bin/contrail-control >>/var/log/contrail/contrail-control.stdout 2>/var/log/contrail/contrail-control.err"],
+	    "command": ["/bin/bash", "-c", "/usr/bin/contrail-control --DISCOVERY.server=127.0.0.1 --DISCOVERY.port=5998 >>/var/log/contrail/contrail-control.stdout 2>/var/log/contrail/contrail-control.err"],
```
Comment (pedro-r-marques) [inline cluster/contrail-control.manifest]: Is the control node now executing correctly if /etc/hosts doesn't have an entry for hostname ?

File: cluster/kube-network-manager.manifest [hunk 3]
```diff
@@ -7,15 +7,24 @@
 	"containers":[{
 	    "name": "kube-network-manager",
 	    "image": "opencontrail/kube-network-manager",
-	    "args": ["--", "--public_net=10.1.4.0/24"],
+	    "command": ["/bin/sh", "-c", "/go/kube-network-manager -- --private_net=10.10.0.0/16 --portal_net=10.0.0.0/16 --public_net=10.1.0.0/16 1>>/var/log/contrail/kube-network-manager.stdout 2>>/var/log/contrail/kube-network-manager.err"],
```
Comment (pedro-r-marques) [inline cluster/kube-network-manager.manifest]: We can't hardcode this stuff...
Lets move this file to kube-network-manager.manifest.j2 and use a jinja template variable.

File: cluster/contrail-control.manifest (hunk 2 above)
Comment (rombie) [inline cluster/contrail-control.manifest]: In all the setups we tried (vagrant and aws), hostname was correctly
resolvable to the container's host's address. Hence we thought, it is
better not to override that.

File: cluster/contrail-api.manifest (hunk 1 above)
Comment (rombie) [inline cluster/contrail-api.manifest]: Sanju proposed that we should move contrail stuff to pods eventually.
Having every component talk to contrail-discovery, makes it easier to move
them around in future. It is also the most tested version in house, where
in we always use discovery in contrail setups during testing.

File: cluster/kube-network-manager.manifest (hunk 3 above)
Comment (rombie) [inline cluster/kube-network-manager.manifest]: ok

File: cluster/contrail-collector.manifest [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+    "apiVersion": "v1",
+    "kind": "Pod",
+    "metadata": {"name":"contrail-coll"},
```
Comment (pedro-r-marques) [inline cluster/contrail-collector.manifest]: contrail-collector

File: cluster/contrail-collector.manifest [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "spec":{
+        "hostNetwork": true,
+        "containers":[{
+            "name": "contrail-coll",
```
Comment (pedro-r-marques) [inline cluster/contrail-collector.manifest]: contrail-collector

File: cluster/contrail-collector.manifest [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            "image": "opencontrail/analytics:2.20",
+            "command": ["/bin/sh", "-c", "/usr/bin/contrail-collector 1>>/var/log/contrail/contrail-collector.stdout 2>>/var/log/contrail/contrail-collector.err"],
+            "ports": [{
+                "name": "contrail-coll",
```
Comment (pedro-r-marques) [inline cluster/contrail-collector.manifest]: No abbreviations.

File: cluster/contrail-query-engine.manifest [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{
+    "apiVersion": "v1",
+    "kind": "Pod",
+    "metadata": {"name":"contrail-qe"},
```
Comment (pedro-r-marques) [inline cluster/contrail-query-engine.manifest]: contrail-query-engine

File: cluster/contrail-query-engine.manifest [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    "spec":{
+        "hostNetwork": true,
+        "containers":[{
+            "name": "contrail-qe",
+            "image": "opencontrail/analytics:2.20",
+            "command": ["/bin/sh", "-c", "/usr/bin/contrail-query-engine 1>>/var/log/contrail/contrail-query-engine.stdout 2>>/var/log/contrail/contrail-query-engine.err"],
+            "ports": [{
+                "name": "contrail-qe",
```
Comment (pedro-r-marques) [inline cluster/contrail-query-engine.manifest]: No abbreviations.

File: cluster/contrail-vrouter-agent.manifest [hunk 9]
```diff
@@ -7,7 +7,7 @@
 	"containers":[{
 	    "name": "contrail-vrouter-agent",
 	    "image": "opencontrail/vrouter-agent:2.20",
-            "privileged": "true"
+        "privileged": "true",
```
Comment (pedro-r-marques) [inline cluster/contrail-vrouter-agent.manifest]: Why the indentation change ?

File: cluster/contrail-collector.manifest (hunk 4 above)
Comment (rombie) [inline cluster/contrail-collector.manifest]: Apparently, there is size limit for the name :(

File: cluster/kube-network-manager.manifest [hunk 10]
```diff
@@ -1,21 +1,40 @@
+{% set params = "" -%}
```
Comment (pedro-r-marques) [inline cluster/kube-network-manager.manifest]: rename file to kube-network-manager.manifest.sls

File: cluster/provision_master.sh [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+# Provision link local service
+function provision_linklocal() {
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" curl -s https://raw.githubusercontent.com/Juniper/contrail-controller/R2.20/src/config/utils/provision_linklocal.py -o /tmp/provision_linklocal.py\"}" | sudo sh'
+    master $cmd
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" python /tmp/provision_linklocal.py --api_server_ip `hostname --ip-address` --api_server_port 8082 --linklocal_service_name kubernetes --linklocal_service_ip 10.0.0.1 --linklocal_service_port 8080 --ipfabric_service_ip `hostname --ip-` --ipfabric_service_port 8080 --oper add\"}" | sudo sh'
+    master $cmd
+}
+
+# Setup kube dns endpoints
+function setup_kube_dns_endpoints() {
```
Comment (pedro-r-marques) [inline cluster/provision_master.sh]: We can put these file in our (Juniper) repository.

File: cluster/provision_master.sh [hunk 12]
```diff
@@ -0,0 +1,130 @@
...
+    master kubectl --namespace=kube-system create -f /etc/kubernetes/addons/kube-ui/kube-ui-endpoint.yaml || true
+    master kubectl --namespace=kube-system create -f /etc/kubernetes/addons/kube-ui/kube-ui-svc-address.yaml || true
+}
+
+# Setup contrail manifest files under kubernetes
+function setup_contrail_manifest_files() {
+    cmd='wget -qO - https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/manifests.hash | grep -v kube-network-manager | grep -v contrail-vrouter-agent | awk "{print \"https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/\"\$1}" | xargs -n1 sudo wget -q --directory-prefix=/etc/contrail/manifests --continue'
```
Comment (pedro-r-marques) [inline cluster/provision_master.sh]: Can't salt copy the manifests directly to the right place ?

File: cluster/contrail-collector.manifest (hunk 4 above)
Comment (rombie) [inline cluster/contrail-collector.manifest]: Apparently, there is/was size limit for the name :( May be they have
relaxed it now

File: cluster/provision_master.sh (hunk 12 above)
Comment (rombie) [inline cluster/provision_master.sh]: Salt can copy, but most of the times docker hangs when it tries to download
all at once. kubernetes also have seen that issue and hence do this. I have
found success in bringing all correctly, only first doing "docker pull",
and then copy the manifest files to /etc/kubernetes/manifests.

https://github.com/Juniper/kubernetes/blob/master/cluster/saltbase/salt/kube-master-addons/kube-master-addons.sh

This also reduces the diffs quite a bit, with just 3 manifest files for
contrail (one for kube-network-manager and 2 for the master and minion
contrail-provision scripts)

**Pull request merged**